torch/csrc/autograd/VariableTypeUtils.h

```diff
@@ -139,6 +139,14 @@ template<typename... Args> inline variable_list flatten_tensor_args(Args&&... ar
 inline Tensor as_view(const Tensor & base, const Tensor & tensor, bool is_bw_differentiable,
         bool is_fw_differentiable, std::function<Tensor(const Tensor&)> view_func=nullptr,
         CreationMeta creation_meta=CreationMeta::DEFAULT, bool allow_tensor_metadata_change=true) {
+  // Note [View of inference tensor]
+  // For inference tensor this code can only be hit outside InferenceMode
+  // since InplaceOrView is in the default_included_set.
+  // If Inplace and View were separate dispatch keys we can just put Inplace
+  // in the default_included_set, so that view ops on inference tensor doesn't
+  // have to go through as_view even outside InferenceMode.
+  if (base.unsafeGetTensorImpl()->is_inference_tensor() || tensor.unsafeGetTensorImpl()->is_inference_tensor())
+    return tensor;
```

Comment:
I'm not sure why you have to test both base and tensor. The invariant is that inference tensors produce inference views, and normal tensors produce normal views, so if base is an inference tensor, it must necessarily be the case that tensor is an inference tensor. I suppose you could add an assert for this case, but I wouldn't test both branches. (This would simplify the tensorlist code below...)